distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/common/downloads/SynchronousClientDownloader.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.test.itests.common.downloads;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.http.nio.client.methods.HttpAsyncMethods;
+
+/**
+ * Provides an asynchronous handle to the synchronous retrieval and download endpoint of a product.
+ */
+class SynchronousClientDownloader extends ClientDownloader {
+    private final InterruptableAsyncCharConsumer consumer = new InterruptableAsyncCharConsumer();
+
+    private final StringBuilder resultsBuilder = new StringBuilder();
+
+    @Override
+    public void startDownload(String metacardId) {
+        client.start();
+        consumer.setBuilder(resultsBuilder);
+        future = client.execute(HttpAsyncMethods.createGet(ClientDownloader.getSyncDownloadUrl(
+                metacardId)), consumer, null);
+    }
+
+    @Override
+    public void stopDownload() {
+        consumer.interrupt();
+        try {
+            client.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Client encountered an error when closing.", e);
+        }
+    }
+
+    @Override
+    protected String processResults() throws ExecutionException {
+        if (future == null) {
+            throw new IllegalStateException("Download was never started, so no future exists");
+        }
+        try {
+            return (String) future.get();
+        } catch (InterruptedException e) {
+            Thread.currentThread()
+                    .interrupt();
+            return processResults();
```

Comment:
Once again, I think this is OK for test scope, but I want to be sure @pklinef (same question as above).